system/usbhcd.c

```diff
@@ -87,6 +88,9 @@ static int num_hcd = 0;
 static int print_row = 0;
 static int print_col = 0;
 
+static const usb_hcd_t *print_hcd = NULL;
+static usb_ep_t *print_ep = NULL;
```

Comment:
You need to initialise these two variables with non-null values in order to generate the necessary relocation records, as is done for `hcd_list` above. This is probably the cause of your crashes.

Reply:
I tried, using intptr_t initially set to -1 and casting accordingly in usb_serial_print() but it is the same crash. Why do print_row and print_col get away with not being null-initialized?

Reply:
Because they contain integer values, not pointer values. The relocation operation needs to be performed on the pointer value stored in the variable, not on the address of the variable.

Reply:
I pushed a commit where I had initialized with a dummy (but valid) non-null pointer. With this I don't see the same crash, but it gets stuck anyway. When I look at it in gdb, it looks like the assignments of print_ep and print_hcd (e.g. in find_attached_usb_keyboard) don't take effect, but it is maybe gdb that looks in the wrong place.